### PR TITLE
Reader Improvements: Update the suggested topics design

### LINKED ISSUE #15964: Reader Improvements: Redesign "Topics you might like"
Show topics using topic chips instead of a list to reduce vertical space taken up by the "Topics you might like" section.

topics-you-might-like-card

### PR DESCRIPTION
**Pull request by emilylaguna**
Fixes #15964

### Notes
- I spent a lot of time today trying to get this to all work in code, but no matter what I did I could not get the constraints to work properly which is why I ended up converting it to a `.xib`. 
- I originally tried to reuse my `TopicsCollectionView` but this is doing something completely different so I ditched that effort

### Screenshots

| Light | Dark |
|:---:|:---:|
|Simulator Screen Shot - iPhone 11 Pro - 2021-02-25 at 18 15 41|Simulator Screen Shot - iPhone 11 Pro - 2021-02-25 at 18 15 39|

### To test:

1. Launch the app
2. Tap on the Reader tab
3. Tap on the 'Discover' tab
4. Locate the 'You Might Like' card
5. You should now see the topics in a horizontally scrolling list
6. Tap on a topic
7. You should be brought to the topic detail view
8. Scroll down a few pages to find a set of topics where the 4th topic appears out of bounds
9. Scroll to make it visible

#### Extra credit testing
1. Open `ReaderTopicsCardCell.swift`
2. Go to line 24
3. Change it to `self.data = data + data + data + data + data`
4. Relaunch the app
5. Verify all the colors loop after the 4th one and doesn't crash

### PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderCardsStreamViewController.swift [hunk 1]
```diff
@@ -32,7 +32,8 @@ class ReaderCardsStreamViewController: ReaderStreamViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         ReaderWelcomeBanner.displayIfNeeded(in: tableView)
-        tableView.register(ReaderTopicsCardCell.self, forCellReuseIdentifier: readerCardTopicsIdentifier)
+
+        tableView.register(ReaderTopicsCardCell.defaultNib, forCellReuseIdentifier: readerCardTopicsIdentifier)
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Reader/ReaderCardsStreamViewController.swift:36]:
```suggestion
        tableView.register(ReaderTopicsCardCell.defaultNib, forCellReuseIdentifier: ReaderTopicsCardCell.defaultReuseID)
```

[nitpick] By making `ReaderTopicsCardCell` conform to `NibReusable` instead, you get `defaultReuseID` for free!

File: WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift [hunk 2]
```diff
@@ -2,36 +2,126 @@ import UIKit
 
 /// A cell that displays topics the user might like
 ///
-class ReaderTopicsCardCell: ReaderTopicsTableCardCell {
-    private let cellIdentifier = "TopicCell"
+class ReaderTopicsCardCell: UITableViewCell, NibLoadable {
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift:5]:
```suggestion
class ReaderTopicsCardCell: UITableViewCell, NibReusable {
```

File: WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift [hunk 3]
```diff
@@ -2,36 +2,126 @@ import UIKit
...
+        contentView.backgroundColor = .clear
     }
 
-    override func cell(forRowAt indexPath: IndexPath, tableView: UITableView, topic: ReaderAbstractTopic?) -> UITableViewCell {
-        guard let tagTopic = topic as? ReaderTagTopic else {
-            return UITableViewCell()
+    private struct Constants {
+        static let title = NSLocalizedString("You might like", comment: "A suggestion of topics the user might like")
+
+        static let reuseIdentifier = ReaderInterestsCollectionViewCell.classNameWithoutNamespaces()
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift]:
```suggestion
```

File: WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift [hunk 4]
```diff
@@ -2,36 +2,126 @@ import UIKit
...
+    }
+}
+
+// MARK: - Collection View: Datasource & Delegate
+extension ReaderTopicsCardCell: UICollectionViewDelegate, UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: Constants.reuseIdentifier,
+                                                            for: indexPath) as? ReaderInterestsCollectionViewCell else {
+            fatalError("Expected a ReaderInterestsCollectionViewCell for identifier: \(Constants.reuseIdentifier)")
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift:64]:
```suggestion
        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ReaderTopicsCardCell.defaultReuseID,
                                                            for: indexPath) as? ReaderInterestsCollectionViewCell else {
            fatalError("Expected a ReaderInterestsCollectionViewCell for identifier: \(ReaderTopicsCardCell.defaultReuseID)")
```

File: WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsCollectionViewCell.swift [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import UIKit
 
-class ReaderInterestsCollectionViewCell: UICollectionViewCell {
+class ReaderInterestsCollectionViewCell: UICollectionViewCell, NibLoadable {
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsCollectionViewCell.swift]:
```suggestion
class ReaderInterestsCollectionViewCell: UICollectionViewCell, NibReusable {
```

File: WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift [hunk 6]
```diff
@@ -2,36 +2,126 @@ import UIKit
...
+
+        applyStyles()
+
+        // Configure header
+        headerLabel.text = Constants.title
+
+        // Configure collection view
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.register(ReaderInterestsCollectionViewCell.defaultNib,
+                                forCellWithReuseIdentifier: Constants.reuseIdentifier)
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift]:
```suggestion
                                forCellWithReuseIdentifier: ReaderInterestsCollectionViewCell.defaultReuseID)
```

**Review by momo-ozawa — APPROVED** [review]
Works as described! (Tested both sections: To test + Extra credit testing)
Just a couple nitpicks about `NibReusable`

File: WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift (hunk 4 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift:64]: Done in: 616998ecec24851a03b947bb70a219a03766ce49

File: WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsCollectionViewCell.swift (hunk 5 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsCollectionViewCell.swift]: Done in: 616998ecec24851a03b947bb70a219a03766ce49

File: WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift (hunk 6 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift]: Done in: 616998ecec24851a03b947bb70a219a03766ce49

File: WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift (hunk 3 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift]: Done in: 616998ecec24851a03b947bb70a219a03766ce49

File: WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift (hunk 2 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Reader/ReaderTopicsCardCell.swift:5]: Done in: 616998ecec24851a03b947bb70a219a03766ce49

File: WordPress/Classes/ViewRelated/Reader/ReaderCardsStreamViewController.swift (hunk 1 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Reader/ReaderCardsStreamViewController.swift:36]: Done in: 616998ecec24851a03b947bb70a219a03766ce49

**Review by momo-ozawa — APPROVED** [review]
👍

**Comment by emilylaguna** [comment]
@momo-ozawa I updated the dark mode colors based on some design review. Can you give another review?

File: WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsStyleGuide.swift [hunk 7]
```diff
@@ -66,3 +66,90 @@ class ReaderInterestsStyleGuide {
...
+    /// The array of colors from the designs
+    /// Note: I am explictly using the MurielColor names instead of using the semantic ones
+    ///       since these are explicit and not semantic colors.
+    static let colors: [TopicStyle] = [
+        // Green
+        .init(textColor: .init(colorName: .green, section: .text),
+              backgroundColor: .init(colorName: .green, section: .background),
+              borderColor: .init(colorName: .green, section: .border)),
+
+        // Blue
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsStyleGuide.swift]: Super small nitpick...
`// Purple` instead of `// Blue`?

**Review by momo-ozawa — APPROVED** [review]
Looking good! 👍 

Light | Dark
--- | ---
Simulator Screen Shot - iPhone 12 Pro - 2021-03-03 at 17 00 36 | Simulator Screen Shot - iPhone 12 Pro - 2021-03-03 at 17 00 30

File: WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsStyleGuide.swift (hunk 7 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsStyleGuide.swift]: Oops, done: add5712e9dd64fd1390663a85c00448d6e339008

**Pull request merged**